src/python/pants/backend/python/util_rules/pex_requirements.py

```diff
@@ -269,6 +273,74 @@ def __bool__(self) -> bool:
         return bool(self.req_strings)
 
 
+@dataclass(frozen=True)
+class ResolvePexConfig:
+    """Configuration from `[python]` that impacts how the resolve is created."""
+
+    constraints_file: tuple[Digest, FileEntry] | None
+
+    @property
+    def constraints_file_path_and_hash(self) -> tuple[str, str] | None:
```

Comment:
It would be better if this returned a dataclass and not a tuple whose unpacking is not self-explanatory. Reading the code that uses this was confusing.

Reply:
But this can wait for a followup.

Reply:
John agreed with you. Will do :)